gramex/data.py

```diff
@@ -884,7 +884,7 @@ def _filter_db(engine, table, meta, controls, args, source='select', id=[]):
                 raise ValueError('_limit not integer: %r' % controls['_limit'])
             query = query.limit(limit)
             meta['limit'] = limit
-        return pd.read_sql(query, engine)
+        return gramex.cache.query(query, engine, state=[table_name])
 
```

Comment:
Could you add a test case please? A test case which would fail in absence of this change.